### PR TITLE
Merge Sharding tests in Go migrated from Python

### PR DESCRIPTION
**Pull request by ajeetj**
- The base structure is copied from resharding test case
- Minor changes in base_sharding.go as we are comparing multiple values now
- updated config for merge sharding

### CODE REVIEW

File: go/test/endtoend/sharding/mergesharding/mergesharding_base.go [hunk 1]
```diff
@@ -0,0 +1,640 @@
...
+	insertTabletTemplateKsID = `insert into %s (parent_id, id, msg, custom_ksid_col) values (%d, %d, '%s', %d) /* vtgate:: keyspace_id:%d */ /* id:%d */`
+
+	// initial shards
+	// range -40, 40-80 & 80-
+	shard0 = &cluster.Shard{Name: "-40"}
+	shard1 = &cluster.Shard{Name: "40-80"}
+	shard2 = &cluster.Shard{Name: "80-"}
+
+	// merge shard
+	// merging -40 & 40-80 t0 -80
```
Comment (deepthi) [inline go/test/endtoend/sharding/mergesharding/mergesharding_base.go]: type: t0 -> to

File: go/test/endtoend/sharding/mergesharding/v3/mergesharding_v3_test.go [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+package v3
+
+import (
+	"testing"
+
+	sharding "vitess.io/vitess/go/test/endtoend/sharding/mergesharding"
+)
+
+// TestV3MergeSharding - tests merge sharding using a INT column
+func TestV3MergeSharding(t *testing.T) {
```
Comment (deepthi) [inline go/test/endtoend/sharding/mergesharding/v3/mergesharding_v3_test.go]: same comment here. remove v3, add comment before boolean.

File: test/config.json [hunk 3]
```diff
@@ -318,6 +296,28 @@
 				"site_test"
 			]
 		},
+		"merge_sharding": {
+			"File": "mergesharding_v3_test.go",
```
Comment (deepthi) [inline test/config.json]: rename v3 -> int

File: go/test/endtoend/sharding/base_sharding.go [hunk 4]
```diff
@@ -183,16 +183,16 @@ func CheckBinlogPlayerVars(t *testing.T, vttablet cluster.Vttablet, sourceShards
 	replicationSourceObj := reflect.ValueOf(tabletVars["VReplicationSource"])
 	replicationSourceValue := []string{}
 
-	assert.Equal(t,
-		fmt.Sprintf("%v", replicationSourceObj.MapKeys()),
-		fmt.Sprintf("%v", reflect.ValueOf(tabletVars["VReplicationSourceTablet"]).MapKeys()))
+	assert.Equal(t, len(replicationSourceObj.MapKeys()), len(reflect.ValueOf(tabletVars["VReplicationSourceTablet"]).MapKeys()))
```
[the remark this answers is not in the document]
Comment (ajeetj) [reply go/test/endtoend/sharding/base_sharding.go]: Turns out the actual keys were just indexes 1,2... so on. And the values are totally different. So if we can also compare the counts here.

File: go/test/endtoend/sharding/mergesharding/string/mergesharding_string_test.go [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+import (
+	"testing"
+
+	sharding "vitess.io/vitess/go/test/endtoend/sharding/mergesharding"
+)
+
+// TestV3MergeShardingString - tests merge sharding using a Byte column
+func TestV3MergeShardingString(t *testing.T) {
+	sharding.TestMergesharding(t, true)
```
[the remark this answers is not in the document]
Comment (ajeetj) [reply go/test/endtoend/sharding/mergesharding/string/mergesharding_string_test.go]: Done

File: go/test/endtoend/sharding/mergesharding/mergesharding_base.go (hunk 1 above)
Comment (ajeetj) [reply go/test/endtoend/sharding/mergesharding/mergesharding_base.go]: Done

File: go/test/endtoend/sharding/mergesharding/v3/mergesharding_v3_test.go (hunk 2 above)
Comment (ajeetj) [reply go/test/endtoend/sharding/mergesharding/v3/mergesharding_v3_test.go]: Updated

File: test/config.json (hunk 3 above)
Comment (ajeetj) [reply test/config.json]: Updated

**Comment by ajeetj** [comment]
@deepthi Addressed review comments. Waiting for checks to pass.

**Pull request merged**